spec/models/services/data_consolidator_spec.rb

```diff
@@ -75,40 +78,50 @@ def self.mapper_class
+    let(:subject) { process_entities_to_cmp(entities) }
 
-    context 'entity is not immutable, no entity has been pushed to connec! before' do
-      it 'returns all external entities' do
+    context 'when entity is mutable' do
+      before do
         allow(BankTransaction).to receive(:immutable?).and_return(false)
-        entities = data_consolidator.consolidate_external_entities(external_entities, 'BankTransaction')
-        ent = process_entities_to_cmp(entities)
-        expect(ent).to eq(external_entities)
+      end
+      context 'when none of the entities have been pushed to connec!' do
+        it 'returns all external entities' do
+          expect(subject).to eq(external_entities)
+        end
       end
     end
 
-    context 'entity is immutable and some entities have been pushed to connec! before' do
-      it 'returns some external entities' do
-        Maestrano::Connector::Rails::IdMap.create(external_id: 'TX8934077941420180731', external_entity: 'transaction', organization_id: organization.id, connec_entity: 'banktransaction', connec_id: 1)
+    context 'when entity is immutable' do
+      before do
         allow(BankTransaction).to receive(:immutable?).and_return(true)
-        entities = data_consolidator.consolidate_external_entities(external_entities, 'BankTransaction')
-        ent = process_entities_to_cmp(entities)
-        expect(ent.count).to eq(2)
-        expect(ent[0]).to eq(external_entities[1])
-        expect(ent[1]).to eq(external_entities[2])
       end
-    end
 
-    context 'entity is immutable and all entities have been pushed to connec! before' do
-      it 'return empty array' do
-        Maestrano::Connector::Rails::IdMap.create(external_id: 'TX8934077941420180731', external_entity: 'transaction', organization_id: organization.id, connec_entity: 'banktransaction', connec_id: 1)
-        Maestrano::Connector::Rails::IdMap.create(external_id: '201801160', external_entity: 'transaction', organization_id: organization.id, connec_entity: 'banktransaction', connec_id: 2)
-        Maestrano::Connector::Rails::IdMap.create(external_id: '201801121', external_entity: 'transaction', organization_id: organization.id, connec_entity: 'banktransaction', connec_id: 3)
-        allow(BankTransaction).to receive(:immutable?).and_return(true)
-        entities = data_consolidator.consolidate_external_entities(external_entities, 'BankTransaction')
-        expect(entities.count).to eq(0)
+      context 'when some entities have been pushed to connec!' do
+        before do
+          create(:idmap, external_id: 'TX8934077941420180731', external_entity: 'transaction', organization_id: organization.id, connec_entity: 'banktransaction', connec_id: 1)
+        end
+
+        it 'only returns new entities' do
+          # expect(subject.count).to eq(2)
+          expect(subject[0]).to eq(external_entities[1])
+          expect(subject[1]).to eq(external_entities[2])
```

Comment:
`match_array` didn't work?
In that case, you should still have an expectation on the count. Otherwise a rotation of the entities will pass the specs ;)